design-documents/graph-ql/coverage/customer-orders.md

```diff
@@ -0,0 +1,196 @@
+
+## Order Type Schema
+
+The proposed type for the customer order might look like:
+
+```graphql
+# order details
+type Order {
+    order_date: String! # date when the order was placed.
+    status: String! # current status of the order.
+    order_number: String! # order number.
+    items: [OrderItem] # collection of all the items purchased
+    order_pricing: OrderPricingInterface! # pricing details for the order.
+    invoices: [Invoice] # invoice list for the order.
+    credit_memos: [CreditMemo] # credit memo list for the order.
+    shipments: [OrderShipment] # shipping list for the order.
+    payment_method: [PaymentMethod] # payment details for the order.
+}
+```
+
+### Order Item
+
+The order items will be presented as separate interface which will have multiple implementations for invoice, shipment and credit memo types.
+
+```graphql
+interface SalesProductInterface {
+    name: String # name of the base product.
+    sku: String! # sku of the base product.
+    url: String # url of the base product.
+    final_price: Float! # final price for the base product including all the child products and selected options.
+    discounts: [Discount] # final discount information for the base product including discounts on options and child products.
+    selected_options: [String!] # selected options for the base product. for e.g color, size etc.
+    entered_options: [KeyValue] # entered option for the base product. for e.g logo image etc.
+}
+```
+
+The `SalesProductInterface` will be implemented by the following types:
+
+```graphql
+# Order Product implementation of OrderProductInterface
+type OrderItem implements SalesProductInterface {
+    ordered: Float! # number of items items
+    shipped: Float! # number of shipped items
+    refunded: Float! # number of refunded items
+    invoiced: Float! # number of invoiced items
+    children: [OrderChildItem]
+}
+
+type OrderChildItem implements SalesProductInterface{
+    ordered: Float! # number of items items
+    shipped: Float! # number of shipped items
+    refunded: Float! # number of refunded items
+    invoiced: Float! # number of invoiced items
+}
+```
+
+### Payment Method Schema
+
+To provide more customization for different payment solutions, the payment method will be represented by own type instead of simple string:
+
+```graphql
+#  payment method used to pay for the order
+type PaymentMethod {
+    name: String! # payment method name for e.g Braintree, Authorize etc
+    type: String! # payment method type used to pay for the order for e.g Credit Card, PayPal etc.
+    additional_data: [KeyValue] # additional data per payment method type
+}
+```
+
+### Pricing Schema
+
+As entities like order, invoice, credit memo might have complex pricing type:
+
+```graphql
+interface OrderPricingInterface {
+    subtotal: Float! # order subtotal excluding, shipping, discounts and tax
+    discounts: [Discount] # all the discounts applied to the order
```

Comment:
```suggestion
    applied_discounts: [Discount] # all the discounts applied to the order
```

Reply:
Does it make sense to rename them? If discounts are presented in order it means they already applied. The same is true for other amounts.